### PR TITLE
Add temperature field

### PR DESCRIPTION
**Pull request by streeve**
Reorder fields during this change to guard against issues seen in other codes 

Increasingly strong motivation to move to `ParticleLists`, but for now just add another field

### CODE REVIEW

File: src/CabanaPD_Particles.hpp [hunk 1]
```diff
@@ -220,6 +220,7 @@ class Particles<MemorySpace, PMB, Dimension>
         auto y = sliceCurrentPosition();
         auto vol = sliceVolume();
         auto nofail = sliceNoFail();
+        auto temp = sliceNoFail();
```
Comment (pabloseleson) [inline src/CabanaPD_Particles.hpp:223]: "sliceNoFail" should be temperature instead.

Comment (streeve) [reply src/CabanaPD_Particles.hpp]: Fixed.

File: src/CabanaPD_Boundary.hpp [hunk 2]
```diff
@@ -168,6 +172,42 @@ struct BoundaryCondition
...
+        auto temp = particles.sliceTemperature();
+        auto x = particles.sliceReferencePosition();
+        auto value = _value;
+        auto index_space = _index_space._view;
+        Kokkos::RangePolicy<ExecSpace> policy( 0, index_space.size() );
+        Kokkos::parallel_for(
+            "CabanaPD::BC::apply", policy, KOKKOS_LAMBDA( const int b ) {
+                auto pid = index_space( b );
+                // This is specifically for the thermal deformation problem
+                temp( pid ) = value * x( pid, 1 ) * t;
```
Comment (pabloseleson) [inline src/CabanaPD_Boundary.hpp:206]: @streeve : We need to replace Line 206 by 

temp( pid ) = value * (x( pid, 1 ) - low_corner[1]) * t;

Could you please make the change to CabanaPD_Boundary.hpp?

Comment (streeve) [reply src/CabanaPD_Boundary.hpp]: done

**Comment by streeve** [comment]
Closing in favor of #93

**Pull request closed without merging**